picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -768,14 +768,22 @@ public Builder executor(@NonNull ExecutorService executorService) {
     }
 
     /** Specify the memory cache used for the most recent images. */
-    public Builder memoryCache(@NonNull Cache memoryCache) {
-      if (memoryCache == null) {
-        throw new IllegalArgumentException("Memory cache must not be null.");
+    public Builder memoryCacheBytes(int maxByteCount) {
+      if (maxByteCount <= 0) {
+        throw new IllegalArgumentException("maxByteCount <= 0: " + maxByteCount);
       }
       if (this.cache != null) {
         throw new IllegalStateException("Memory cache already set.");
       }
-      this.cache = memoryCache;
+      this.cache = new LruCache(maxByteCount);
+      return this;
+    }
+
+    public Builder noMemoryCache() {
+      if (this.cache != null) {
+        throw new IllegalStateException("Memory cache already set.");
```

Comment:
left this around for now, but I'd like to remove these checks for a newBuilder nicety.